packages/components/src/sites-table/use-sites-table-filtering.tsx

```diff
@@ -43,10 +45,16 @@ export function useSitesTableFiltering(
 
 		for ( const status of statuses ) {
 			status.count = filteredByStatus[ status.name ].length;
+			if ( ! showHidden ) {
+				filteredByStatus[ status.name ] = [ ...filteredByStatus[ status.name ] ].filter(
+					( site ) => site.visible
+				);
+				status.hiddenCount = status.count - filteredByStatus[ status.name ].length;
+			}
```

Comment:
This is a little bit hard to read but I can't think of a more readable version.

Reply:
I think we don't need to destruct the array: 
```diff
- filteredByStatus[ status.name ] = [ ...filteredByStatus[ status.name ] ].filter(
+ filteredByStatus[ status.name ] = filteredByStatus[ status.name ].filter(
```